File: Easy/1256-rank-transform-of-an-array/rank-transform-of-an-array.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        n=len(arr)

        temp=[]
        mp={}
       
        for a in arr:
            if a not in mp:
                temp.append(a)
                mp[a]=0

        temp.sort()

        for i, num in enumerate(temp):
            mp[num]=i+1

        return [mp[a] for a in arr]

File: Easy/1256-rank-transform-of-an-array/rank-transform-of-an-array.java
class Solution{
    public int[] arrayRankTransform(int[] arr){
        int n=arr.length;

        ArrayList<Integer> temp=new ArrayList<>();
        HashMap<Integer, Integer> mp=new HashMap<>();

        for(int a: arr){
            if(!mp.containsKey(a)){
                temp.add(a);
                mp.put(a, 0);
            }
        }

        Collections.sort(temp);

        for(int i=0; i<temp.size(); i++){
            mp.put(temp.get(i), i+1);
        }

        for(int i=0; i<n; i++){
            arr[i]=mp.get(arr[i]);
        }

        return arr;
    }
}